app/src/main/java/dev/arkbuilders/drop/app/data/repository/NetworkStatusImpl.kt

```diff
@@ -45,17 +41,12 @@ class NetworkStatusImpl(
     }
 
     private fun checkIsOnline(): Boolean {
-        val network: Network = cm.activeNetwork ?: return false
+        val network: Network = connectivityManager.activeNetwork ?: return false
         val networkCapabilities: NetworkCapabilities =
-            cm.getNetworkCapabilities(network) ?: return false
-
-        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
-            networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET) &&
-                networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_VALIDATED) &&
-                networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_NOT_SUSPENDED)
-        } else {
-            networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET) &&
-                networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_VALIDATED)
-        }
+            connectivityManager.getNetworkCapabilities(network) ?: return false
+
+        return networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET) &&
+            networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_VALIDATED) &&
+            networkCapabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_NOT_SUSPENDED)
     }
```

Comment:
Let’s keep the version checks and lower the minimum SDK to 26 like in Rate

Reply:
Alright